src/core_modules/capture-core/components/TemplateSelector/TemplateSelector.container.tsx

```diff
@@ -1,17 +1,18 @@
-// @flow
 import React from 'react';
 import { TemplateSelector as TemplateSelectorComponent } from './TemplateSelector.component';
 import { useNavigate, buildUrlQueryString, useLocationQuery } from '../../utils/routing';
 import { useTEITemplates, useProgramStageTemplates } from './hooks';
+import type { WorkingListTemplate } from './workingListsBase.types';
 
 export const TemplateSelector = () => {
     const { navigate } = useNavigate();
     const { programId, orgUnitId } = useLocationQuery();
-    const { TEITemplates, loading: loadingTEITemplates } = useTEITemplates(programId);
-    const { programStageTemplates, loading: loadingProgramStageTemplates } = useProgramStageTemplates(programId);
+    const { TEITemplates, loading: loadingTEITemplates } = useTEITemplates(programId || '');
+    const { programStageTemplates, loading: loadingProgramStageTemplates } = useProgramStageTemplates(programId || '');
```

Comment:
I would rather change `useTEITemplates` and `useProgramStageTemplates` to take in either a string or undefined, and then only do the network request if the ids are defined. I don't think we need the `|| ''` here.